library/network_connections.py

```diff
@@ -1156,7 +1156,10 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_802_1X_DOMAIN_SUFFIX_MATCH,
                     connection["ieee802_1x"]["domain_suffix_match"],
                 )
-
+        if connection["match"]:
+            if connection["match"]["path"] is not None:
```

Comment:
According to the unit tests, `connection["match"]["path"]` is never None, so this would be always True. Should this be

```suggestion
            if connection["match"]["path"]:
```

?

Reply:
`match.path` setting should allow empty list, maybe we should directly remove  `if connection["match"]["path"] is not None:`

Reply:
I agree with Till here. If the path can be an empty list or None, then doing `if connection["match"]["path"]:` seems better.

Reply:
`match.path` setting should allow empty list, then when `connection["match"]["path"]` is `[]`, `if connection["match"]["path"]` was evaluated to false, then the property can not be set. As Till mentioned, "`connection["match"]["path"]` is never None, so this would be always True". Maybe this condition is redundant, let us remove it.